internal/kubernetes/operator/features/crds.go

```diff
@@ -0,0 +1,203 @@
+package features
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	urlTemplate                 = "https://raw.githubusercontent.com/mongodb/mongodb-atlas-kubernetes/v%s/bundle/manifests/atlas.mongodb.com_%s.yaml"
+	maxDepth                    = 100
+	requestTimeout              = 10 * time.Second
+	ResourceVersion             = "app.kubernetes.io/version"
+	ResourceAtlasProject        = "atlasprojects"
+	ResourceAtlasDeployment     = "atlasdeployments"
+	ResourceAtlasDatabaseUser   = "atlasdatabaseusers"
+	ResourceAtlasBackupSchedule = "atlasbackupschedules"
+	ResourceAtlasBackupPolicy   = "atlasbackuppolicies"
+	ResourceAtlasTeam           = "atlasteams"
+)
+
+var (
+	ErrVersionNotSupportedFmt    = "version '%s' is not supported"
+	ErrDownloadResourceFailedFmt = "can not download resource '%s': %v"
+	ErrDocumentIsEmpty           = errors.New("document is empty")
+	ErrDocumentHasNoVersions     = errors.New("document contains no versions")
+	ErrDocumentHasNoSchema       = errors.New("document contains no Schema")
+	ErrDocumentHasNoSpec         = errors.New("document contains no Spec")
+
+	VersionsToResourcesMap = map[string][]string{
+		"1.4.1": {
+			ResourceAtlasProject,
+			ResourceAtlasDeployment,
+			ResourceAtlasDatabaseUser,
+			ResourceAtlasBackupSchedule,
+			ResourceAtlasBackupPolicy,
+		},
+		"1.5.0": {
+			ResourceAtlasDatabaseUser,
+			ResourceAtlasProject,
+			ResourceAtlasDeployment,
+			ResourceAtlasBackupSchedule,
+			ResourceAtlasBackupPolicy,
+			ResourceAtlasTeam,
+		},
+		"1.5.1": {
+			ResourceAtlasDatabaseUser,
+			ResourceAtlasProject,
+			ResourceAtlasDeployment,
+			ResourceAtlasBackupSchedule,
+			ResourceAtlasBackupPolicy,
+			ResourceAtlasTeam,
+		},
+	}
+)
+
+func GetSupportedVersions() []string {
```

Comment:
getters are non ideomatic
```suggestion
func SupportedVersions() []string {
```

Reply:
Fixed